src/lazy_index_map.rs

```diff
@@ -13,6 +13,7 @@ use smallvec::SmallVec;
 pub struct LazyIndexMap<K, V> {
     vec: SmallVec<[(K, V); 8]>,
     map: OnceLock<AHashMap<K, usize>>,
+    last_find: Arc<Mutex<usize>>,
```

Comment:
@davidhewitt I'm interested if you think there's something faster than `Arc<Mutex<...>>` we can use where while still satisfying the requirement of implementing `Sync`?

Reply:
If `last_find` is not a problem for correctness, could just use `UnsafeCell`? E.g. if worst case is that the field isn't found from `last_find` to the end, and then the start is searched, it's probably suitable.

Reply:
(using `UnsafeCell` you'll need to wrap it like `struct LastFind(UnsafeCell<usize>)` and `unsafe impl Sync for LastFind`.)

Reply:
Also `AtomicUsize` is the safe option which will still be much faster than a mutex and on some platforms maybe no different to `UnsafeCell`, might want the `portable_atomic` crate as a dependency given the range of platforms this might be packaged on.